src/debot/term_encryption_box.rs

```diff
@@ -0,0 +1,235 @@
+        Ok(nacl_box(
+            self.client.clone(),
+            ParamsOfNaclBox {
+                decrypted: data.clone(),
+                nonce: self.nonce.clone(),
+                their_public: self.their_pubkey.clone(),
+                secret: self.secret.clone(),
+            },
+        )
+        .unwrap()
+        .encrypted)
+    }
+    async fn decrypt(&self, data: &String) -> ClientResult<String> {
+        Ok(nacl_box_open(
+            self.client.clone(),
+            ParamsOfNaclBoxOpen {
+                encrypted: data.clone(),
+                nonce: self.nonce.clone(),
+                their_public: self.their_pubkey.clone(),
+                secret: self.secret.clone(),
+            },
+        )
+        .unwrap()
+        .decrypted)
+    }
+}
+
+pub(super) struct TerminalEncryptionBox {
+    handle: EncryptionBoxHandle,
+}
+
+impl TerminalEncryptionBox {
+    pub async fn new(params: ParamsOfTerminalEncryptionBox) -> Result<Self, String> {
+        let key: String;
+
+        {
+            let stdio = io::stdin();
+            let mut reader = stdio.lock();
+            let mut writer = io::stdout();
+            let enter_str = "enter seed phrase or path to keypair file";
+            let value = input(enter_str, &mut reader, &mut writer);
+            let pair = load_keypair(&value).map_err(|e| e.to_string())?;
+            key = format!("{:064}", pair.secret);
+        }
+
+        let registered_box = match params.box_type {
+            EncryptionBoxType::SecretNaCl => {
+                register_encryption_box(
+                    params.context.clone(),
+                    NaClSecretBox {
+                        key: key,
+                        nonce: params.nonce,
+                        client: params.context.clone(),
+                    },
+                )
+                .await
+                .map(|r| r.handle)
+                .unwrap_or(EncryptionBoxHandle(0))
+            }
+            EncryptionBoxType::NaCl => {
+                register_encryption_box(
+                    params.context.clone(),
+                    NaClBox {
+                        their_pubkey: params.their_pubkey,
+                        secret: key,
+                        nonce: params.nonce,
+                        client: params.context.clone(),
+                    },
+                )
+                .await
+                .map(|r| r.handle)
+                .unwrap_or(EncryptionBoxHandle(0))
+            }
+            EncryptionBoxType::ChaCha20 => {
+                register_encryption_box(
+                    params.context.clone(),
+                    ChaChaBox {
+                        key: key,
+                        nonce: params.nonce,
+                        client: params.context.clone(),
+                    },
+                )
+                .await
+                .map(|r| r.handle)
+                .unwrap_or(EncryptionBoxHandle(0))
```

Comment:
Move this 2 lines to the caller function and write here: .await?;

Reply:
done